### PR TITLE
Pass sourcesets using TaskInputs.files, not .file

### PR DESCRIPTION
**Pull request by zpencer**
TaskInputs.file can only accept real files, not sourcesets:
https://discuss.gradle.org/t/gradle-4-3-rc1-is-available-for-testing/24415/16

### CODE REVIEW

File: src/test/groovy/com/google/plugins/ProtobufAndroidPluginTest.groovy [hunk 1]
```diff
@@ -55,6 +55,6 @@ buildscript {
 
     where:
     androidPluginVersion << ["2.2.0", "2.2.0", "3.0.0-alpha7"]
-    gradleVersion << ["2.14.1", "3.0", "4.1-milestone-1"]
+    gradleVersion << ["2.14.1", "3.0", "4.1-milestone-1", "4.3-rc-2"]
```
Comment (zhangkun83) [inline src/test/groovy/com/google/plugins/ProtobufAndroidPluginTest.groovy]: The 4.1 milestone should probably be changed to 4.1

File: src/main/groovy/com/google/protobuf/gradle/Utils.groovy [hunk 2]
```diff
@@ -63,7 +63,7 @@ class Utils {
 
   static void addFilesToTaskInputs(Project project, TaskInputs inputs, Object files) {
     if (compareGradleVersion(project, "3.0") >= 0) {
-      inputs.file(files).skipWhenEmpty()
+      inputs.files(files).skipWhenEmpty()
```
Comment (zhangkun83) [inline src/main/groovy/com/google/protobuf/gradle/Utils.groovy]: Wow, glad to see it can be fixed in such a clean way.

**Comment by zhangkun83** [comment]
LGTM! Thanks @zpencer!

File: src/main/groovy/com/google/protobuf/gradle/Utils.groovy (hunk 2 above)
Comment (teodor-appd) [reply src/main/groovy/com/google/protobuf/gradle/Utils.groovy:66]: Had to dig as deep as this diff to solve my Value `org.gradle.api.internal.tasks.DefaultTaskOutputs@1cc16f4b' specified for property '$1' cannot be converted to a file`. Thanks a lot!

Also, if you know how to debug such a problem (or how did you do it), please tell me.

**Pull request merged**